src/oom/oomd-manager.c

```diff
@@ -807,6 +820,10 @@ int manager_start(
         if (r < 0)
                 return r;
 
+        r = sd_event_add_exit(m->event, NULL, clean_prekills, m->prekill_ctxs);
+        if (r < 0)
+                return r;
```

Comment:
We don't need this since we already store all prekills in the `Manager` object since they will be freed when the `Manager` object is freed.

Reply:
I moved the cleanup in manager_free() and I see it's being called on quit:
```
systemd-oomd[311761]: Cleaning up unfinished prekill hook state for cgroup /system.slice/run-p311776-i311777.service
```